src/algorithms/noop.ts

```diff
@@ -34,9 +37,49 @@ export class NoopAlgorithm extends AbstractAlgorithm {
     markers,
     map,
     mapCanvasProjection,
+    forceRecalculate,
   }: AlgorithmInput): AlgorithmOutput {
     return {
-      clusters: this.cluster({ markers, map, mapCanvasProjection }),
+      clusters: this.cluster({
+        markers,
+        map,
+        mapCanvasProjection,
+        forceRecalculate,
+      }),
+      changed: false,
+    };
+  }
+
+  protected cluster(input: AlgorithmInput): Cluster[] {
+    return this.noop(input);
+  }
+}
+
+/**
+ * NoopViewport algorithm does not generate any clusters but it does filter markers by the an extended viewport.
+ */
+export class NoopViewportAlgorithm extends AbstractViewportAlgorithm {
```

Comment:
What about changing the "Noop" part of the name?

"filter markers by the an extended viewport" does not sound like a no-op?